stock_location_content_template/models/stock_location_content_check.py

```diff
@@ -0,0 +1,242 @@
+    )
+    line_ids = fields.One2many(
+        "stock.location.content.check.line", "check_id", string="Products"
+    )
+    picking_ids = fields.One2many("stock.picking", "check_id", string="Pickings")
+
+    @api.model
+    def create(self, vals):
+        vals.update(
+            {
+                "name": self.env["ir.sequence"].next_by_code(
+                    "stock.location.content.check"
+                )
+            }
+        )
+        return super(StocklocationContentCheck, self).create(vals)
+
+    def action_confirm(self):
+        for rec in self:
+            rec.state = "confirmed"
+
+    def action_complete(self):
+        for rec in self:
+            rec._create_out_pick()
+            rec._create_internal_pick()
+            rec.state = "completed"
+
+    def _create_out_pick(self):
+        picking_obj = self.env["stock.picking"]
+        location_id = self.location_id
+        location_dest_id = self.env["stock.location"].search(
+            [("usage", "=", "customer")], limit=1
+        )
+        wh_id = location_id.get_warehouse()
+        pick_type_id = wh_id.out_type_id
+        for rec in self:
+            move_lines = []
+            for line in rec.line_ids:
+                if line.current_qty >= (line.current_qty - line.counted_qty) > 0:
+                    move_lines.append(
+                        (
+                            0,
+                            0,
+                            {
+                                "name": line.product_id.name,
+                                "product_id": line.product_id.id,
+                                "product_uom": line.product_id.uom_id.id,
+                                "product_uom_qty": line.current_qty - line.counted_qty,
+                                "picking_type_id": pick_type_id.id,
+                                "location_id": location_id.id,
+                                "location_dest_id": location_dest_id.id,
+                            },
+                        )
+                    )
+            if move_lines:
+                picking_obj.create(
+                    {
+                        "picking_type_id": pick_type_id.id,
+                        "location_id": location_id.id,
+                        "location_dest_id": location_dest_id.id,
+                        "move_lines": move_lines,
+                        "check_id": rec.id,
+                    }
+                )
+
+    def _create_internal_pick(self):
+        picking_obj = self.env["stock.picking"]
+        for rec in self:
+            move_lines_self_parent = []
+            move_lines_parent_self = []
+            if rec.location_id.location_id.usage == "view":
+                raise UserError(_("You can't use view type location in transfer."))
+            wh_id = rec.location_id.get_warehouse()
+            pick_type_id = wh_id.int_type_id
+            for line in rec.line_ids:
+                if not line.replenished_qty == 0.0:
```

Comment:
As said for fields definitions, you need to use digit precision and float_compare here.